src/llanalyzer/Manager.cc

```diff
@@ -0,0 +1,285 @@
+		}
+
+	// Dump Analyzer Set
+	if (analyzer_set)
+		analyzer_set->DumpDebug();
+#endif
+	}
+
+bool Manager::EnableAnalyzer(const Tag& tag)
+	{
+	Component* p = Lookup(tag);
+
+	if ( ! p )
+		return false;
+
+	DBG_LOG(DBG_LLANALYZER, "Enabling analyzer %s", p->Name().c_str());
+	p->SetEnabled(true);
+
+	return true;
+	}
+
+bool Manager::EnableAnalyzer(EnumVal* val)
+	{
+	Component* p = Lookup(val);
+
+	if ( ! p )
+		return false;
+
+	DBG_LOG(DBG_LLANALYZER, "Enabling analyzer %s", p->Name().c_str());
+	p->SetEnabled(true);
+
+	return true;
+	}
+
+bool Manager::DisableAnalyzer(const Tag& tag)
+	{
+	Component* p = Lookup(tag);
+
+	if ( ! p )
+		return false;
+
+	DBG_LOG(DBG_LLANALYZER, "Disabling analyzer %s", p->Name().c_str());
+	p->SetEnabled(false);
+
+	return true;
+	}
+
+bool Manager::DisableAnalyzer(EnumVal* val)
+	{
+	Component* p = Lookup(val);
+
+	if ( ! p )
+		return false;
+
+	DBG_LOG(DBG_LLANALYZER, "Disabling analyzer %s", p->Name().c_str());
+	p->SetEnabled(false);
+
+	return true;
+	}
+
+void Manager::DisableAllAnalyzers()
+	{
+	DBG_LOG(DBG_LLANALYZER, "Disabling all analyzers");
+
+	std::list<Component*> all_analyzers = GetComponents();
+	for ( const auto& analyzer : all_analyzers )
+		analyzer->SetEnabled(false);
+	}
+
+zeek::llanalyzer::Tag Manager::GetAnalyzerTag(const char* name)
+	{
+	return GetComponentTag(name);
+	}
+
+bool Manager::IsEnabled(Tag tag)
+	{
+	if ( ! tag )
+		return false;
+
+	Component* p = Lookup(tag);
+
+	if ( ! p )
+		return false;
+
+	return p->Enabled();
+	}
+
+bool Manager::IsEnabled(EnumVal* val)
+	{
+	Component* p = Lookup(val);
+
+	if ( ! p )
+		return false;
+
+	return p->Enabled();
+	}
+
+Analyzer* Manager::InstantiateAnalyzer(const Tag& tag)
+	{
+	Component* c = Lookup(tag);
+
+	if ( ! c )
+		{
+		reporter->InternalWarning("request to instantiate unknown llanalyzer");
+		return nullptr;
+		}
+
+	if ( ! c->Enabled() )
+		return nullptr;
+
+	if ( ! c->Factory() )
+		{
+		reporter->InternalWarning("analyzer %s cannot be instantiated dynamically", GetComponentName(tag).c_str());
+		return nullptr;
+		}
+
+	Analyzer* a = c->Factory()();
+
+	if ( ! a )
+		{
+		reporter->InternalWarning("analyzer instantiation failed");
+		return nullptr;
+		}
+
+	if ( tag != a->GetAnalyzerTag() )
+		{
+		reporter->InternalError("Mismatch of requested analyzer %s and instantiated analyzer %s. This usually means that the plugin author made a mistake.",
+		                        GetComponentName(tag).c_str(), GetComponentName(a->GetAnalyzerTag()).c_str());
+		return nullptr;
+		}
+
+	return a;
+	}
+
+Analyzer* Manager::InstantiateAnalyzer(const std::string& name)
+	{
+	Tag tag = GetComponentTag(name);
+	return tag ? InstantiateAnalyzer(tag) : nullptr;
+	}
+
+void Manager::ProcessPacket(Packet* packet)
+	{
+#ifdef DEBUG
+	static size_t counter = 0;
+	DBG_LOG(DBG_LLANALYZER, "Analyzing packet %ld, ts=%.3f...", ++counter, packet->time);
+#endif
+
+	if ( ! analyzer_set )
```

Comment:
Can this (legitimately) happen?

Reply:
No. Should we turn that into an assertion or remove it?